arch/cortex-m4/src/mpu.rs

```diff
@@ -116,23 +118,55 @@ impl kernel::mpu::MPU for MPU {
         }
     }
 
-    fn set_mpu(&self,
-               region_num: u32,
-               start_addr: u32,
-               len: PowerOfTwo,
-               subregion_mask: u8,
-               execute: kernel::mpu::ExecutePermission,
-               access: kernel::mpu::AccessPermission) {
+    fn create_region(region_num: usize,
```

Comment:
Doing checking on region_num and returning `None` if it is greater than 8 would be good.